middleware/webhooks.js

```diff
@@ -1,31 +1,30 @@
 const crypto = require('crypto');
-const shopStore = require('../shopStore');
 
-const { SHOPIFY_APP_SECRET } = process.env;
+module.exports = function createWithWebhook({ secret, shopStore }) {
+  return function withWebhook(request, response, next) {
+    const { body: data } = request;
+    const hmac = request.get('X-Shopify-Hmac-Sha256');
+    const topic = request.get('X-Shopify-Topic');
+    const shopDomain = request.get('X-Shopify-Shop-Domain');
 
-module.exports = function withWebhook(request, response, next) {
-  const { body: data } = request;
-  const hmac = request.get('X-Shopify-Hmac-Sha256');
-  const topic = request.get('X-Shopify-Topic');
-  const shopDomain = request.get('X-Shopify-Shop-Domain');
+    const generated_hash = crypto
+      .createHmac('sha256', SHOPIFY_APP_SECRET)
+      // probably change the parser we're using
```

Comment:
can you open an issue instead of this comment?

Reply:
The comment was already there I just moved it, but yeah it should be an issue instead.